### PR TITLE
Update kubernetes demo/config for example

### PR DESCRIPTION
**Pull request by mattjmcnaughton**
A couple of changes to make the kubernetes config work
and follow best practices.

- Use `deployment` instead of `pod`.
- Use correct image names.

After these changes, running `kubectl create -f demo/config/` creates a
working deployment/service.

### CODE REVIEW

File: demo/README.md [hunk 1]
```diff
@@ -20,11 +22,14 @@ docker build -t <some-registry>/hugo hugo/
 docker push <some-registry>/hugo <some-registry>/git-sync
 ```
 
+If you end up using a different registry, be sure to update `config/pod.yaml`.
```
Comment (stp-ip) [inline demo/README.md]: s/pod/deployment/

File: demo/config/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    matchLabels:
+      name: blog
+  template:
+    metadata:
+      labels:
+        name: blog
+    spec:
+      containers:
+      - name: git-sync
+        image: staging-k8s.gcr.io/git-sync-amd64:v2.0.5
```
Comment (stp-ip) [inline demo/config/deployment.yaml]: The latest version would be v2.0.6. Any reason why you used .5?

Comment (mattjmcnaughton) [reply demo/config/deployment.yaml]: Nope, no reason! Other than I think I thought 2.0.5 was the most recent version haha...

File: demo/README.md (hunk 1 above)
Comment (mattjmcnaughton) [reply demo/README.md]: Great catch :)

**Comment by mattjmcnaughton** [comment]
@stp-ip thanks for the feedback - I made your changes so this is ready for another look :)

**Comment by mattjmcnaughton** [comment]
Yup!

**Pull request merged**